library/socket/basicsocket/local_address_spec.rb

```diff
@@ -0,0 +1,170 @@
+require_relative '../spec_helper'
+require_relative '../fixtures/classes'
+
+def each_basic_socket_type
+  describe 'using TCPSocket' do
+    before do
+      @s = TCPServer.new('127.0.0.1', 0)
+      @a = TCPSocket.new('127.0.0.1', @s.addr[1])
+      @b = @s.accept
+    end
+    after do
+      [@b, @a, @s].each(&:close)
+    end
+
+    it 'uses AF_INET as the address family' do
+      @la.afamily.should == Socket::AF_INET
+    end
+
+    it 'uses PF_INET as the protocol family' do
+      @la.pfamily.should == Socket::PF_INET
+    end
+
+    it 'uses SOCK_STREAM as the socket type' do
+      @la.socktype.should == Socket::SOCK_STREAM
+    end
+
+    it 'uses the correct IP address' do
+      @la.ip_address.should == '127.0.0.1'
+    end
+
+    it 'uses a different port then the server' do
+      @la.ip_port.should != @s.addr[1]
+    end
+
+    it 'equals remote_address of peer socket' do
+      @la.to_s.should == @b.remote_address.to_s
+    end
+
+    yield
+  end
+
+  guard -> { SocketSpecs.ipv6_available? } do
+    describe 'using IPv6' do
```

Comment:
I think it should be possible to use `SocketSpecs.each_ip_protocol` here, could you try that?